docs/framework/react/guide/server-functions.md

```diff
@@ -0,0 +1,245 @@
+})
+
+// Call it like this:
+yourFn(123);
+```
+
+Only one argument is supported, but you can pass an object if you need to pass multiple values:
+
+```typescript
+const yourFn = createServerFn('POST', async (obj: { a: number, b: number }) => {
+  // Server-side code lives here
+})
+```
+
+Any serializable value can be passed as an argument to a server function.
+
+# Serialization
+
+Anything that can be serialized via `JSON.stringify` and `JSON.parse` can be used as an argument to or returned from a
+server function.
+
+This includes:
+
+- `string`
+- `number`
+- `boolean`
+- `null`
+- `Array`
+- `Object`
+
+In addition, we support serializing the following:
+
+- [`ReadableStream`](https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream)s of other serializable types
+- [`Promise`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise)s of other
+  serializable types
+
+# No-JS Server Functions
+
+Without JavaScript enabled, there's only one way to execute server functions: by submitting a form.
+
+This is done by adding a `form` element to the page
+with [the HTML attribute `action`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/action).
+
+> Notice that we mentioned the **HTML** attribute `action`. This attribute only accepts a string in HTML, just like all
+> other attributes.
+>
+> While React
+> 19 [added support for passing a function to `action`](https://react.dev/reference/react-dom/components/form#form),
+> it's
+> a React-specific feature and not part of the HTML standard.
+
+The `action` attribute tells the browser where to send the form data when the form is submitted. In this case, we want
+to send the form data to the server function.
+
+To do this, we can utilize the `url` property of the server function:
+
+```typescript
+const yourFn = createServerFn('POST', async () => {
+  // Server-side code lives here
+})
+
+console.log(yourFn.url)
+```
+
+And pass this to the `action` attribute of the form:
+
+```tsx
+function Component() {
+  return (
+    <form action={yourFn.url} method="POST">
+      <button type="submit">Click me!</button>
+    </form>
+  )
+}
+```
+
+When the form is submitted, the server function will be executed.
+
+## No-JS Server Function Arguments
+
+To pass arguments to a server function when submitting a form, you can use the `input` element with the `name` attribute
+to attach the argument to the [`FormData`](https://developer.mozilla.org/en-US/docs/Web/API/FormData) passed to your
+server function:
+
+```tsx
+const yourFn = createServerFn('POST', async (formData: FormData) => {
+  // `val` will be '123'
+  const val = formData.get('val');
+  // ...
+})
+
+function Component() {
+  return (
+    //  We need to tell the server that our data type is `multipart/form-data` by setting the `encType` attribute on the form.
+    <form action={yourFn.url} method="POST" encType="multipart/form-data">
+      <input type="hidden" name="val" value="123" />
```

Comment:
I'd add a callout on how client-side fields like this should not be "trusted". Purely for security due-dilligence.

Reply:
Oh! I didn't even realize `type="hidden"` was there. Removing!